cli/module_graph.rs

```diff
@@ -353,6 +353,29 @@ impl ModuleGraphLoader {
     let mut types_directives = vec![];
     let mut type_headers = vec![];
 
+    // IMPORTANT: source_file.url might be different than requested
+    // module_specifier - this situation needs to be handled manually
+    if module_specifier.to_string() != source_file.url.to_string() {
+      // TODO(bartlomieju): refactor, this is a band-aid
+      self.graph.0.insert(
```

Comment:
There must be at least a paragraph here explaining what's going on and how to fix it in the next PR. If not don't land